src/AutoMerge/juliaup.jl

```diff
@@ -0,0 +1,73 @@
+using juliaup_jll: juliaup, julia
+
+const available_julia_versions = VersionNumber[]
+const version_re = r"^(\d+\.\d+\.\d+(-[\w\d]+)?)$"
+
+function find_available_julia_versions()
```

Comment:
this looks like it could be a little fragile if `juliaup list` changes its printing, but I don't think there's a better way to do it now. Maybe we could ask for `juliaup list --json` or something to have a stable machine-readable format. But I don't think that needs to block this feature.